### ISSUE TITLE
Support for LaTeX text

### ISSUE THREAD

**Issue by nipunbatra** [vega/vega#898]
Is it possible to display LaTeX text in Vega? Can we use MathJax to render the math?

cross-ref: https://github.com/vega/vega-lite/issues/2640

**Comment by jheer** [comment]
Thanks for the suggestion (and links)! I agree that it would be nice to be able to support this. However, Vega currently relies on the basic text rendering features of Canvas and SVG, while the output of the web-based LaTeX renderers I've looked at require a great deal of custom formatting (sometimes at the per-character level). For example, the linked plot.ly example uses a number of custom SVG glyph definitions, and also exhibits some errors (namely, showing LaTeX markup rather than rendered output in the interactive tooltips). In short, making this work "natively" in Vega would be non-trivial and could require substantial modification of the Vega scenegraph. So this will not be on our short-term roadmap.

In the meantime, there are some possible alternative routes to consider, though I don't think they are particularly promising:
- I could imagine trying to pre-process latex markup to generate plain unicode text, and then using that text as the labels within Vega. However, I'd worry about necessary custom formatting getting lost in the process. Nevertheless, it might be worth experimenting to see how far this goes. Complex math with fractions, etc might not be supported, but perhaps other items would work.
- Adding rendered-LaTeX to a Vega view as a post-processing step. The vega-tooltip library (which I believe uses HTML rendering) might be able to support LaTeX. Though hack-ish, one might also perform post-processing on SVG output from Vega to add rendered LaTeX.

**Comment by nipunbatra** [comment]
@jheer Many thanks for the detailed explanation. I was highly ignorant on how much effort this would require!

**Comment by davidanthoff** [comment]
It would be really great to have LaTeX support, for figures in scientific publications it seems quite key.

I googled around a bit more and found some plans over at KaTeX for an [intermediate](https://github.com/Khan/KaTeX/issues/376) representation that could potentially power various rendering backends. That sounds potentially useful for vega? Maybe it would make sense to try to coordinate a bit with them, so that whatever they are creating could be used in vega?

**Comment by davidanthoff** [comment]
It might also be worth taking another look at MathJax. v3.0 is not yet out, but my understanding is that they have much more of a pipeline design, where one can for example swap out output formats more easily. So one thing it for example supports is converting latex into [svg](http://docs.mathjax.org/en/latest/output/svg.html) directly, and if I understand the description correctly that SVG should be relatively self-contained. The new design might also make it feasible to add a new, custom output processor (not sure about that part, though).

**Comment by tmori3y2** [comment]
Related issue: https://github.com/vega/vega/issues/1289#issuecomment-538179411

I made a demo comparing Tex and MathML's JsonML notation.

https://observablehq.com/@tmori3y2/jsonml-notation-of-mathml-by-katex

Both notations should be converted to svg with MathJax.

However, MathJax's toMML is not working on observable, so I made MathML with Katex. (Katex does not support MathML input)

In conclusion, parsing inline tex nontation seems better than JsonML.

**Comment by tmori3y2** [comment]
> However, MathJax's toMML is not working on observable, so I made MathML with Katex. (But Katex does not support MathML input...)
> 

I misunderstood that Katex outputs SVG, but it does not support it.

Since it is output in span, it is necessary to import css.

I don't know whether span or svg is better for vega.

**Comment by davidanthoff** [comment]
MathJax 3 is out now, maybe worth revisiting whether it could be used somehow?

**Comment by isentropic** [comment]
Any progress on this? I want to try implementing Vega through vegalite.jl into Plots.jl package, but this is holding me back before even starting

**Comment by rben01** [comment]
Since MathJax 3 can render to SVG, it seems like it's at least theoretically possible to add LaTeX support to Vega, as it's possible to include one SVG in another and to render an SVG to a canvas.

**Comment by RoyiAvital** [comment]
It seems the landscape has somewhat changed since @jheer 's analysis.  
Any chance reevaluating this considering the updates in MathJaX and KaTeX?

**Comment by smorad** [comment]
We were planning to move to Altair for our publications, but no mathjax/latex/katex support is preventing us from moving. We very often need fractions and other non-unicode symbols in the labels.

Is there any workaround to get math rendering in Altair?

**Comment by domoritz** [comment]
It would be interesting to try to pass the code through to the SVG and then replace it with the rendered latex. You'd probably need to manually set label sizes since this breaks automatic spacing. Of course super hacky but a step to seeing what a solution could look like.
